lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,513 @@
+import { CurrencyInstance } from '../db/types';
+import Logger from '../Logger';
+import swapErrors from '../swaps/errors';
+import SwapClient, {
+  ChannelBalance,
+  ClientStatus,
+  PaymentState,
+  WalletBalance,
+  TradingLimits,
+} from '../swaps/SwapClient';
+import { SwapDeal } from '../swaps/types';
+import { UnitConverter } from '../utils/UnitConverter';
+import { ConnextWallet } from './ConnextWallet';
+import errors, { errorCodes } from './errors';
+import {
+  Channel,
+  ConnextChannelCount,
+  ConnextClientConfig,
+  ConnextInfo,
+  ConnextVersion,
+  TokenPaymentRequest,
+  TokenPaymentResponse,
+} from './types';
+
+const MAX_AMOUNT = Number.MAX_SAFE_INTEGER;
+
+/**
+ * A class representing a client to interact with connext.
+ */
+class ConnextClient extends SwapClient {
+  public readonly type = SwapClientType.Connext;
+  public readonly finalLock = 100;
+  public address?: string;
+  /** A map of currency symbols to token addresses. */
+  public tokenAddresses = new Map<string, string>();
+  private network: string;
+  private ethProviderUrl: string;
+  private nodeUrl: string;
+  private unitConverter: UnitConverter;
+
+  private channel: IConnextClient | undefined;
+  private wallet: ConnextWallet | undefined;
+
+  /**
+   * Creates a connext client.
+   */
+  constructor({
+    config,
+    logger,
+    unitConverter,
+    currencyInstances,
+  }: {
+    unitConverter: UnitConverter;
+    config: ConnextClientConfig;
+    currencyInstances: CurrencyInstance[],
+    logger: Logger;
+  }) {
+    super(logger, config.disable);
+
+    this.network = config.network || '';
+    this.ethProviderUrl = config.ethProviderUrl || '';
+    this.nodeUrl = config.nodeUrl || '';
+    this.unitConverter = unitConverter;
+    this.setTokenAddresses(currencyInstances);
+  }
+
+  public get minutesPerBlock() {
+    return 0.25; // 15 seconds per block target
+  }
+
+  public get label() {
+    return 'Connext';
+  }
+
+  /**
+   * Checks for connectivity and gets our Connext account address
+   */
+  public initSpecific = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    if (this.wallet) {
+      await this.initConnext();
+      this.setStatus(ClientStatus.Initialized);
+      this.emit('initialized');
+      await this.verifyConnectionWithTimeout();
+    }
+  }
+  /**
+   * Initiates wallet for the Connext client
+   */
+  public initWallet = async (seedMnemonic: string[]) => {
+    this.wallet = new ConnextWallet(seedMnemonic.join(' '));
+  }
+
+  public initConnext = async () => {
+    if (!this.wallet) {
+      throw errors.CONNEXT_WALLET_NOT_INITIATED;
+    }
+    const wallet = this.wallet;
+    const channel = await connext.connect(this.network, {
+      ethProviderUrl: this.ethProviderUrl,
+      nodeUrl: this.nodeUrl,
+      xpub: wallet.xpub,
+      keyGen: (index: string) => wallet.keyGen(index),
+      store: new ConnextStore(new FileStorage()),
```

Comment:
If I'm reading the code correctly it will create a data directory to `./connext-store` in the runtime directory of xud? We could provide a path to xud's data directory here from config.

Reply:
Correct, this can be configured into the FileStorage class. Where would you like to assign the directory?

Reply:
If we go with the REST API wrapper approach then just having it configurable via runtime arguments or config file would do.

Reply:
Correct, this is no longer relevant 👍